### DISCUSSION TITLE
Why does &_expand=toppings not work?

### DISCUSSION THREAD

**Discussion by DavidEdwardDarden** [nss-day-cohort-47/C47-Discussions#70] [Help Me] [answered]
In apiManager.js... under the getSingleSnack function... I have:

return fetch(`${apiURL}/snacks/${snackId}?_expand=type&_expand=shape&_expand=inFlavor&_expand=season`)
	.then(response => response.json())
----------------------------------------------------------------------------
BUT if I add: &_expand=toppings  it does not work... and I do not understand why?
I tried a lot of other things... but it would be much easier for me if this worked.
I'm temporarily at a halt! (I'm on step 4... correct me if I'm wrong but you don't really have to code anything for number three... it's just about putting the relationship correctly in the ERD for now)

**Comment by DavidEdwardDarden** [comment]
I also tried to put the topping name into an array if it appeared in snacks[{"toppings": "string" ...... and then joining them with a comma with .join into a new variable. And making it an html representation It got overly complicated and I have become temporarily delayed..

I could hard code in the list of toppings... and work from there but that seems like cheating.

**Comment by D-Upchurch** [comment]
with the toppings you are looking for _embed=snackToppings I believe, though I couldn't 100% explain why that is, I have it at the end of the first fetch call like so "&_expand=inFlavor&_embed=snackToppings`" I then have a second function to expand toppings because it didn't like me embedding snackToppings and expanding toppings in the same line. Hope this helps! I don't have mine fully pulling those in yet myself but I've seen it working from that point

**Comment by Chadwell-Clark** [comment]
snackToppings is the join table for snacks and toppings, it allows the many to many relationship between snacks and Toppings. Many toppings can be on one snack and many snacks may have the same topping. How to best use this information is still not fully fleshed out in my mind yet

**Comment by BryanNilsen** [comment, marked as answer]
This is one of the limitations with json-server... you're trying to access related data in a many-to-many relationship, which is currently not supported with their built-in functionality - (that I know of).

Ultimately, you will have to make two requests for data... one to the snacks table, expanding the related (one-to-many) data (season, flavor, type, and shape)

To get the toppings, (the many-to-many relationship) you'll need to target the snackToppings table, using the ID of the snack that you want the toppings for, and then use _expand to get the related topping info.
